domain/controllerconfig/state/state.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+	"database/sql"
+	"github.com/juju/errors"
+	jujucontroller "github.com/juju/juju/controller"
+	"github.com/juju/juju/domain"
+)
+
+// State represents a type for interacting with the underlying state.
+type State struct {
+	*domain.StateBase
+}
+
+// NewState returns a new State for interacting with the underlying state.
+func NewState(factory domain.DBFactory) *State {
+	return &State{
+		StateBase: domain.NewStateBase(factory),
+	}
+}
+
+// ControllerConfig ...
+func (st *State) ControllerConfig(ctx context.Context) (jujucontroller.Config, error) {
+	return nil, nil
+}
+
+// UpdateControllerConfig allows changing some of the configuration
+// for the controller. Changes passed in updateAttrs will be applied
+// to the current config, and keys in removeAttrs will be unset (and
+// so revert to their defaults). Only a subset of keys can be changed
+// after bootstrapping.
+func (st *State) UpdateControllerConfig(ctx context.Context, updateAttrs map[string]interface{}, removeAttrs []string) error {
```

Comment:
Interested to see how we can use [this feature](https://github.com/canonical/sqlair/pull/48) when it lands.